File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        cheak={0:-1}
        total=0
        for indx,el in enumerate(nums):
            total += el
            rem = total %k
            if rem not in cheak:
                cheak[rem]=indx
            elif indx-cheak[rem]>1:
                return True
        return False
        

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        HashMap<Integer,Integer> mp=new HashMap<>();
        mp.put(0,-1);
        int total=0;
        for(int i=0;i<nums.length;i++){
            total+=nums[i];
            int rem=total%k;
            if(!mp.containsKey(rem)){
                mp.put(rem,i);
            }else if((i-mp.get(rem))>=2){
                 return true; 
            }
        }
        return false;
    }
}